lib/concurrent-ruby/concurrent/utility/processor_counter.rb

```diff
@@ -113,6 +118,26 @@ def compute_cpu_quota
           end
         end
       end
+
+      def compute_cpu_shares
+        if RbConfig::CONFIG["target_os"].include?("linux")
+          if cgroups_v2?
+            # Ref: https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2#phase-1-convert-from-cgroups-v1-settings-to-v2
+            weight = File.read('/sys/fs/cgroup/cpu.weight').to_f
```

Comment:
It's probably fine, but in `compute_cpu_quota` we only ever read files we tested for existence prior.

Here we check `/sys/fs/cgroup/cpu.max` and assume it means `/sys/fs/cgroup/cpu.weight` exists? Maybe that's a correct assertion, but I personally don't know if it's true.

I think ultimately it would be simpler to not extract `cgroups_v1? / cgroups_v2?` and check if `/sys/fs/cgroup/cpu.weight` exists here.

Reply:
Agreed, this seems safer